src/test/java/com/shop/frontend/stepdefinitions/Stepdefs.java

```diff
@@ -126,6 +128,12 @@ public void user_should_be_navigated_to_the_products_page() {
         assertEquals(expectedUrl, actualUrl, "All products button is not working.");
     }
 
+    // Author: Priyanka
+    @And("Click on All tab")
+    public void clickOnAllTab() {
+        productPage.clickOnTabAll();
```

Comment:
productPage object is not instantiated anywhere. This step definition will not compile successfully. In the main branch it is defined in the setUp method, but not here. Have you updated your branch with mains content before creating pull request?